### PR TITLE
Feature/login page

### PR DESCRIPTION
**Pull request by kristenmb**
## *Purpose:*
*What does this merge do? Check all that apply.*
- [x] Adds a new feature
- [x] Improves an existing feature
- [x] Fixes a bug
- [ ] Cleans up (logs, tests, etc.)

## *Description: What does this PR do?*
*Briefly, but clearly describe what you did, and explain any new code.*
Adds login, signup, and logout functionality. 
Adjusts routing of app to include a landing page that will display all active tours and allow user to select a tour to view more details on
Updates tests to log in and out user for each round of tests
Adds styling to login and sign up pages
Includes a conditional render on each component that redirects to login if there is no user logged in. 

## *Instructions for reviewer:*
*Where should the reviewer start?*
*How should this be manually tested?*
*Any background context you want to provide?*
*What are the relevant tickets?*

## *Issues:*
*Are there any concerns, issues, or bugs in this branch? If so describe them.*
- the not found component no longer seems to work... it just redirects to dashboard if the page isn't found... not a huge bug but something to look into

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have used tests to prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my change

### CODE REVIEW

File: cypress/integration/navbar_spec.js [hunk 1]
```diff
@@ -1,71 +1,86 @@
...
+
+    cy.url().should('contain', '/contacts')
+  });
+
+  it('should have an icon in the add to calendar link', () => {
+    cy.get('.NavBar a svg:last').should('have.attr', 'data-icon', 'address-book')
+  });
+
+  after('Logout', () => {
+    cy.get('.Header').find('.logout-btn').click()
```
Comment (DrewBradley) [inline cypress/integration/navbar_spec.js:84]: Wow! Way to be thorough! Killer job!

File: src/components/landingPage/LandingPage.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          <p className='tour-dates'>dd/mm/yyyy - dd/mm/yyyy</p>
+        </div>
+      </section>
+    </>
+  )
+}
+
+/* I am imagining we will add the user query and or mutation(creation) in the handle login function, and then this component will mount with all the tours listed. When a tour is selected it can set a global/context variable with the correct tour info/tell which info needs to be loaded on the dashboard, which will then display the appropriate info.  */
+
+export default LandingPage
```
Comment (DrewBradley) [inline src/components/landingPage/LandingPage.js:38]: This looks gorgeous!

**Review by DrewBradley — APPROVED** [review]
Amazing work Kristen! You are the greatest!

File: src/components/App.js [hunk 3]
```diff
@@ -8,29 +8,33 @@ import Form from './form/Form'
...
 
 function App() {
   const { appLoading, appError } = useApp();
   const { currentUser } = useAuth()
 
   return (
     <div className="App">
-      <Header />
+      {currentUser && <Header />}
+      {!currentUser && <Redirect to='/login'/>}
```
Comment (richardltyler) [inline src/components/App.js:25]: I'm liking the redirect here!

File: src/components/header/Header.js [hunk 4]
```diff
@@ -1,14 +1,28 @@
...
 import logo from '../../assets/CuratourLogo.png'
 import './Header.css';
 import { Link } from 'react-router-dom'
+import { useAuth } from '../../context/AuthContext';
 
 const Header = () => {
+  const { signOut, currentUser } = useAuth()
+
+
+  async function handleLogout(event) {
```
Comment (richardltyler) [inline src/components/header/Header.js:11]: good eye! Logout functionality hadn't occurred to me

**Review by richardltyler — APPROVED** [review]
This looks awesome!

**Comment by richardltyler** [comment]
For documentation's sake: 

I'm merging this branch even though it is failing the tests for CI. Travis CI has multiple open issues regarding the reason this test is failing. Namely [this issue](https://github.com/cypress-io/cypress/issues/7062). Basically, `cy.visit()` is not being understood by Travis CI's servers so all tests fail. I pulled down this branch and ran all tests locally, and they all pass.

**Pull request merged**